src/anyp/TrafficMode.h

```diff
@@ -67,18 +68,94 @@ class TrafficMode
      * Indicating the following are required:
      *  - decryption of CONNECT request
      *  - URL translation from relative to absolute form
-     *  - authentication prohibited on unwrapped requests (only on the CONNECT tunnel)
+     *  - Squid authentication prohibited on unwrapped requests (only on the CONNECT tunnel)
      *  - encrypted outbound server connections
      *  - peer relay prohibited. TODO: re-encrypt and re-wrap with CONNECT
      */
     bool tunnelSslBumping = false;
 
-    /** true if the traffic is in any way intercepted
-     *
-     */
-    bool isIntercepted() { return natIntercept||tproxyIntercept ;}
+    PortKind portKind; ///< the parsed port type value
```

Comment:
Please do not leave this data member uninitialized. Instead of adding some bogus `noPort` value, I recommend adding an explicit single-parameter constructor to force creators to specify the right port type (and then propagate the changes as required to fix the build, fixing, for example, add_http_port()).